qutip/solver/flimesolve.py

```diff
@@ -0,0 +1,646 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jun 29 22:07:34 2023
+
+@author: Fenton
+"""
+
+__all__ = [
+    "flimesolve",
+    "FLiMESolver",
+]
+
+import numpy as np
+from qutip.core import data as _data
+from qutip import Qobj, QobjEvo, operator_to_vector
+from .mesolve import MESolver
+from .integrator import qutip_integrator
+from .solver_base import Solver
+from .result import Result
+from time import time
+from ..ui.progressbar import progress_bars
+from qutip.solver.floquet import fsesolve, FloquetBasis
+
+
+def _floquet_rate_matrix(floquet_basis,
+                         Nt,
+                         T,
+                         c_ops,
+                         c_op_rates,
+                         args,
+                         time_sense=0):
+    '''
+    Parameters
+    ----------
+    floquet_basis : FloquetBasis Object
+        The FloquetBasis object formed from qutip.floquet.FloquetBasis
+    Nt : Int
+        Number of points in one period of the Hamiltonian
+    c_ops : list
+        list of collapse operator matrices as Qobjs
+    c_op_rates : list
+        List of collapse operator rates/magnitudes.
+    args : *dictionary*
+        dictionary of parameters for time-dependent Hamiltonians and
+        collapse operators. ***FIX THIS NEXT BIT*** Currently, the first
+        element MUST be the frequency-dependence of the Hamiltonian
+    time_sense : float 0-1,optional
+        the time sensitivity or secular approximation restriction of 
+        FLiMESolve. Decides "acceptable" values of (frequency/rate) for rate 
+        matrix entries. Lower values imply rate occurs much faster than
+        rotation frtequency, i.e. more important matrix entries. Higher
+        values meaning rates cause changes slower than the Hamiltonian rotates,
+        i.e. the changes average out on longer time scales.
+
+    Returns
+    -------
+    total_R_tensor : 2D Numpy matrix
+        This is the 2D rate matrix for the system,created by summing the
+            Rate matrix of each individual collapse operator. Something
+            something Rate Matrix something something linear Operator.
+    '''
+    Hdim = len(floquet_basis.e_quasi)  # Dimensionality of the Hamiltonian
+    args_key = list(args.values())[0]  # Frequency dependence of Hamiltonian
```

Comment:
Perhaps we can pass in `T` instead of `args` and then set the frequency for rescaled to `1 / T` or similar?

Reply:
P.S. `T` is also missing from the docstrings.